margarine/kde.py

```diff
@@ -73,10 +82,16 @@ def __init__(self, theta, weights, **kwargs):
         self.n = (np.sum(weights)**2)/(np.sum(weights**2))
         theta_max = np.max(theta, axis=0)
         theta_min = np.min(theta, axis=0)
-        a = ((self.n-2)*theta_max-theta_min)/(self.n-3)
-        b = ((self.n-2)*theta_min-theta_max)/(self.n-3)
-        self.theta_min = b
-        self.theta_max = a
+        self.theta_min = kwargs.pop('theta_min', None)
+        self.theta_max = kwargs.pop('theta_max', None)
+        if (self.theta_max is None) or np.any(self.theta_max < theta_max):
+            warnings.warn('Estimating theta_max from samples')
+            a = ((self.n-2)*theta_max-theta_min)/(self.n-3)
+            self.theta_max = a
+        if (self.theta_min is None) or np.any(self.theta_min > theta_min):
+            warnings.warn('Estimating theta_min from samples')
+            b = ((self.n-2)*theta_min-theta_max)/(self.n-3)
+            self.theta_min = b
```

Comment:
Hi @DeaglanBartlett, thanks for having a look at this! It is a great addition to the code. I think to simplify the modification it might be better to leave the calculation of `a` and `b` in and then define `self.theta_min = kwargs.pop('theta_min', b)` and equivalent for `theta_max`?

Reply:
Hi @htjb I agree that would simplify things. I've updated my branch to incorporate your recommendation.